src/aliceVision/matching/ArrayMatcher_hnswlib.hpp

```diff
@@ -137,52 +137,52 @@ class ArrayMatcher_hnswlib : public ArrayMatcher<Scalar, Metric>
     bool SearchNeighbours(const Scalar* query, int nbQuery, IndMatches* pvec_indices,
                           std::vector<DistanceType>* pvec_distances, size_t NN)
     {
-        if(HNSWmatcher.get() == nullptr)
+      if(HNSWmatcher.get() == nullptr)
+      {
+        return false;
+      }
+
+      pvec_distances->resize(nbQuery * NN);
+      pvec_indices->resize(nbQuery * NN);
+
+      #ifdef ALICEVISION_IS_DEFINED
+      #pragma omp parallel for
+      #endif
+      for(int i = 0; i < nbQuery; i++)
```

Comment:
```suggestion
      for(int i = 0; i < nbQuery; ++i)
```
nitpicking but is alway better to prefer preincrement in these cases.